test/testcafe/spec/ChallengeAuthenticatorPhone_spec.js

```diff
@@ -0,0 +1,271 @@
+import phoneVerificationSMSThenVoiceOptions from '../../../playground/mocks/data/idp/idx/authenticator-verification-phone-options';
+import phoneVerificationVoiceThenSMSOptions from '../../../playground/mocks/data/idp/idx/authenticator-verification-phone-options-voice-primary';
+import phoneVerificationVoiceOnlyOption from '../../../playground/mocks/data/idp/idx/authenticator-verification-phone-options-voice-only';
+import smsVerification from '../../../playground/mocks/data/idp/idx/authenticator-verification-phone-sms';
+import voiceVerification from '../../../playground/mocks/data/idp/idx/authenticator-verification-phone-voice';
+import success from '../../../playground/mocks/data/idp/idx/success';
+import invalidCode from '../../../playground/mocks/data/idp/idx/error-email-verify';
+
+const logger = RequestLogger(/challenge|challenge\/resend|challenge\/answer/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+const smsPrimaryMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(phoneVerificationSMSThenVoiceOptions)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge')
+  .respond(smsVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/resend')
+  .respond(smsVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(success);
+
+const voicePrimaryMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(phoneVerificationVoiceThenSMSOptions)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge')
+  .respond(voiceVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(success);
+
+const voiceOnlyMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(phoneVerificationVoiceOnlyOption);
+
+const invalidCodeMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(smsVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(invalidCode, 403);
+
+fixture(`Challenge Phone Form`);
+
+async function setup(t) {
+  const challengeFactorPageObject = new ChallengeFactorPageObject(t);
+  challengeFactorPageObject.navigateToPage();
+  return challengeFactorPageObject;
+}
+
+test
+  .requestHooks(smsPrimaryMock)(`SMS primary mode - has the right labels`, async t => {
+    const challengeFactorPageObject = await setup(t);
+    const pageTitle = challengeFactorPageObject.getPageTitle();
+    const pageSubtitle = challengeFactorPageObject.getFormSubtitle();
+    const primaryButtonText = challengeFactorPageObject.getTextContent('.phone-authenticator-challenge__button--primary');
+    const secondaryButtonText = challengeFactorPageObject
+      .getTextContent('.phone-authenticator-challenge__button--secondary');
+    await t.expect(pageTitle).contains('Verify with your phone');
+    await t.expect(pageSubtitle).contains('Send a code via SMS to');
```

Comment:
wouldn't you verify the entire content ? eg. `send a code via SMS to +1 xxx-xxx-xxxx`